editor/src/components/canvas/dom-walker.ts

```diff
@@ -280,25 +300,47 @@ interface DomWalkerProps {
   scale: number
   onDomReport: (
     elementMetadata: ReadonlyArray<ElementInstanceMetadata>,
-    cachedTreeRoots: Array<ElementPath>,
+    cachedPaths: Array<ElementPath>,
   ) => void
   canvasRootElementElementPath: ElementPath
   validRootPaths: Array<ElementPath>
   mountCount: number
+  canvasInteractionHappening: boolean
 }
 
 export function useDomWalker(props: DomWalkerProps): React.Ref<HTMLDivElement> {
   const containerRef = React.useRef<HTMLDivElement>(null)
   const rootMetadataInStateRef = useRefEditorState(
     (store) => store.editor.domMetadata as ReadonlyArray<ElementInstanceMetadata>,
   )
-  const invalidatedSceneIDsRef = React.useRef<Set<string>>(emptySet())
+  const invalidatedPathsRef = React.useRef<Set<string>>(emptySet()) // For invalidating specific paths only
+  const invalidatedScenesRef = React.useRef<Set<string>>(emptySet()) // For invalidating entire scenes and everything below them
   const invalidatedPathsForStylesheetCacheRef = React.useRef<Set<string>>(emptySet())
   const [initComplete, setInitComplete] = useInvalidateInitCompleteOnMountCount(props.mountCount)
-  const resizeObserver = useResizeObserver(invalidatedSceneIDsRef)
-  const mutationObserver = useMutationObserver(invalidatedSceneIDsRef)
+  const selectedViewsRef = React.useRef(props.selectedViews)
+  const canvasInteractionHappeningRef = React.useRef(props.canvasInteractionHappening)
+
+  if (selectedViewsRef.current !== props.selectedViews) {
+    selectedViewsRef.current = props.selectedViews
```

Comment:
I think there's no reason not to just always assign the value

Reply:
Good point! I'll fix this.